pllug-presentation-system/presentation-data/domdocumentdivider.cpp

```diff
@@ -0,0 +1,125 @@
+#include "domdocumentdivider.h"
+
+const int cMaxNumOfElements = 5;
+
+std::unique_ptr<Presentation> DomDocumentDivider::import(const QByteArray &inputText)
+{
+    QByteArray htmlText = "<root>" + inputText + "</root>";
+    QDomDocument htmlDocument;
+    htmlDocument.setContent(htmlText);
+
+    QVector<QDomDocument> separatedParts = divideBySeparators(htmlDocument);
+    QVector<Slide> undividedSlides = divideByHeaders(separatedParts);
+
+    return std::make_unique<Presentation>(*divideSlides(undividedSlides));
+}
+
+bool DomDocumentDivider::isSeparator(const QDomElement &element)
+{
+    return element.tagName() == "hr";
+}
+
+bool DomDocumentDivider::isHeader(const QDomElement &element)
+{
+    return element.tagName() == "h1";
+}
+
+QVector<QDomDocument> DomDocumentDivider::divideBySeparators(const QDomDocument &htmlDocument)
+{
+    QVector<QDomDocument> rParts;
+    QDomDocument currentPart;
+
+    QDomNode domNode = htmlDocument.documentElement().firstChild();
+    QDomElement domElement;
```

Comment:
We do not use domElement outside the while, I think it's better to move it into while scope.